packages/console/src/pages/SignInExperience/components/ManageLanguageModal/LanguageNav.tsx

```diff
@@ -1,29 +1,25 @@
-import { LanguageTag } from '@logto/language-kit';
-
-import Button from '@/components/Button';
-import Plus from '@/icons/Plus';
+import { languages, LanguageTag, languageTagGuard } from '@logto/language-kit';
 
+import AddLanguageSelector from './AddLanguageSelector';
 import LanguageItem from './LanguageItem';
 import * as style from './LanguageNav.module.scss';
 
 type Props = {
   languageTags: LanguageTag[];
   selectedLanguageTag: LanguageTag;
+  onAddLanguage: (languageTag: LanguageTag) => void;
   onSelect: (languageTag: LanguageTag) => void;
 };
 
-const LanguageNav = ({ languageTags, selectedLanguageTag, onSelect }: Props) => {
-  // TODO: LOG-4146 Add Custom Language
+const LanguageNav = ({ languageTags, selectedLanguageTag, onAddLanguage, onSelect }: Props) => {
+  const languageOptions = Object.entries(languages)
+    .map(([languageTag]) => languageTagGuard.parse(languageTag))
+    .filter((languageTag) => !languageTags.includes(languageTag));
```

Comment:
just `Object.keys`? Looks like the value is not being used.

Reply:
Also

<img width="519" alt="image" src="https://user-images.githubusercontent.com/36393111/192933625-46c33fd8-3ec6-4bc2-8b3e-dbd1875f29c8.png">


Why do we need to guard the language Keys again here?  Looks like the guard itself using the languages lib as the data source.

Reply:
Updated with `Object.keys`.
But failed to infer the key type of `languages`; guarded manually.

<img width="587" alt="image" src="https://user-images.githubusercontent.com/10806653/192944152-f7c66d68-ae80-4fb2-a540-2a49feb8027c.png">

Reply:
if I recall properly, the filter method allows you to assert the returned instance type.  Better run redundant logic just for inferring a type.  Should be a better way.

Reply:
Good call, updated.